pkg/cosign/verify.go

```diff
@@ -1049,14 +1054,12 @@ func VerifyRFC3161Timestamp(sig oci.Signature, tsaCerts *x509.CertPool) (*timest
 		tsBytes = rawSig
 	}
 
-	err = tsaverification.VerifyTimestampResponse(ts.SignedRFC3161Timestamp, bytes.NewReader(tsBytes), tsaCerts)
-	if err != nil {
-		return nil, fmt.Errorf("unable to verify TimestampResponse: %w", err)
-	}
-	acceptedTimestamp := ts
-
-	// FIXME: tsaverification.VerifyTimestampResponse has done this parsing; we shouldn’t parse again.
-	return timestamp.ParseResponse(acceptedTimestamp.SignedRFC3161Timestamp)
+	return tsaverification.VerifyTimestampResponse(ts.SignedRFC3161Timestamp, bytes.NewReader(tsBytes),
```

Comment:
Do we expect users to pass the three certificates or it could be alright with a single leave certificate.

Reply:
If you only pass a leaf, then we can't verify the chain. You must pass at least a trust root - The TSACertificate is optional since it may come from the timestamp itself, and intermediates are optional depending on your PKI.

Reply:
True! We mentioned to verify the chain 👍🏻 .